state/indexer/query_range.go

```diff
@@ -96,14 +96,13 @@ func (qr QueryRange) UpperBoundValue() interface{} {
 func LookForRangesWithHeight(conditions []syntax.Condition) (queryRange QueryRanges, indexes []int, heightRange QueryRange) {
 	queryRange = make(QueryRanges)
 	for i, c := range conditions {
-		heightKey := false
 		if IsRangeOperation(c.Op) {
+			heightKey := c.Tag == types.BlockHeightKey || c.Tag == types.TxHeightKey
```

Comment:
Sorry, I know this was my proposed change but maybe we can delete it from this line and just leave it as is (it will be set in line 106). I am worried I am missing a scenario where we set it here where for some reason the rest of the return values are not set and then there is some unexpected behaviour afterwards. 

The initial bug was due to always setting it to false in L99 (which we are not doing anymore). Do you mind verifying this makes sense?

Reply:
To be honest, I didn't completely understand your solution initially. I just got that the heightKey check should be after the "if IsRangeOperaction(c.Op)". That's why I put it inside the check.

Based on my understanding of the code the "heightKey" must be TRUE only if the current processing operation is "height" or "block", otherwise it must be FALSE. That's why "heightKey" must be initialized on the beginning of each loop, because it is possible to process an operation that is not "heightKey". 

For example, if there are three operations - "height", "account.number", "height" then on first iteration the "heightKey" must be TRUE, then FALSE and finally TRUE again. Does it make sense or I'm missing something?

Reply:
Actually you are right. If you were to not set the `heightKey` every time, it would not set the `heightRange` values correctly. If it were not for that, and we needed to set only `queryRange` it would be fine. Your solution is good the way it is now, thank you for the clarifications. 

This indexer  code is quite tricky and has a bunch of corner cases so it is very easy to get lost in all the possibilities.